src/3d/qgs3dmapscene.cpp

```diff
@@ -314,16 +314,14 @@ void Qgs3DMapScene::onCameraChanged()
     mEngine->camera()->lens()->setOrthographicProjection( -viewWidthFromCenter, viewWidthFromCenter, -viewHeightFromCenter, viewHeightFromCenter, mEngine->camera()->nearPlane(), mEngine->camera()->farPlane() );
   }
 
+  // The near and far planes are computed before the scene update
+  // to get a first approximation needed by `updateScene`.
+  // Then, `updateScene` is called. The active nodes may have changed.
+  // This can affect the scene depth.
+  // Therefore, once the scene is ready, `updateCameraNearFarPlanes`
+  // is called one more time by `setSceneState` to get an accurate result.
+  updateCameraNearFarPlanes();
   updateScene();
-  bool changedCameraPlanes = updateCameraNearFarPlanes();
-
-  if ( changedCameraPlanes )
-  {
-    // repeat update of entities - because we have updated camera's near/far planes,
-    // the active nodes may have changed as well
-    updateScene();
-    updateCameraNearFarPlanes();
-  }
```

Comment:
I think this is not very clear to me. Before we had 1. updateScene(), 2. updateCameraNearFarPlanes(), then optionally followed by 1+2 again. This logic may be outdated now, after having everything handled as chunked entities, and with introduction of QgsChunkedEntity::getNearFarPlaneRange(), we can also account for entities with no active nodes).

Now we would have 1. updateCameraNearFarPlanes(), 2. updateScene(). I would agree with the new logic, that we first get approximate near/far planes and then update scene based on that. But why remove the final updateCameraNearFarPlanes() ? The updateScene() call may have activated some nodes, giving a chance for better near/far plane estimate...?

Reply:
Removed.